op-node/p2p/sync.go

```diff
@@ -211,9 +213,10 @@ type SyncClient struct {
 	// inFlight requests are not repeated
 	inFlight map[uint64]*atomic.Bool
 
-	requests       chan rangeRequest
-	peerRequests   chan peerRequest
-	inFlightChecks chan inFlightCheck
+	rangeRequests       chan rangeRequest
+	activeRangeRequests map[uint64]bool
```

Comment:
Do we need a mutex to protect concurrent read/write of the `activeRangeRequests`? Or should I use `atomic.Bool` instead of vanilla `bool` type?

Reply:
Added in https://github.com/ethereum-optimism/optimism/pull/10063/commits/b4a7863b657a7d34b1c7a0d46c352a278eab0526